libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -104,6 +104,60 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const* _messageType)
+{
+	string functionName =
+		string(_assert ? "assert_helper" : "require_helper") +
+		(_messageType ? ("_" + _messageType->identifier()) : "");
+
+	solAssert(!_assert || !_messageType, "Asserts can't have messages!");
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		if (!_messageType)
+			return Whiskers(R"(
+				function <functionName>(condition) {
+					if iszero(condition) { <invalidOrRevert> }
+				}
+			)")
+			("invalidOrRevert", _assert ? "invalid()" : "revert(0, 0)")
+			("functionName", functionName)
+			.render();
+
+
+		solUnimplemented("require() with two parameters is not yet implemented.");
+		// TODO The code below is completely untested as we don't support StringLiterals yet
```

Comment:
Is this code now duplicated? Couldn't it just use `conversionFunctionSpecial`?

Reply:
The code here is encoding the event signature - conversionFunctionSpecial converts a string literal to a memory string.